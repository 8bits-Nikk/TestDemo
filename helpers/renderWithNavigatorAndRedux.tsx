import React from 'react';
import 'react-native';
import createSagaMiddleware from 'redux-saga';
import {applyMiddleware, createStore, Store} from 'redux';
import rootReducer from '../src/App.reducers';
import root from '../src/App.sagas';
import {render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-redux';

export function renderWithNavigatorAndRedux(
  renderComponent: any,
  reduxEvent?: (store: Store) => void,
) {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
  sagaMiddleware.run(root);

  if (reduxEvent) {
    reduxEvent(store);
  }

  return render(
    <NavigationContainer>
      <Provider store={store}>{renderComponent}</Provider>
    </NavigationContainer>,
  );
}
