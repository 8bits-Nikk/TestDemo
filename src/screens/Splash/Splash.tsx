import {StackActions, useNavigation} from '@react-navigation/native';
import React, {useEffect} from 'react';
import {Image, StyleSheet, View} from 'react-native';
import {imgChat} from '../../assets/assets';
import asyncStorage from '../../service/async-storage/asyncStorage';

const Splash = () => {
  const navigation = useNavigation();
  useEffect(() => {
    asyncStorage.getLoginStatus().then(response => {
      if (response === null) {
        setTimeout(() => {
          navigation.dispatch(StackActions.replace('LoginScreen'));
        }, 1500);
      } else {
        setTimeout(() => {
          navigation.dispatch(StackActions.replace('HomeScreen'));
        }, 1500);
      }
    });
  }, [navigation]);

  return (
    <View style={styles.body}>
      <Image
        source={imgChat}
        style={styles.img}
        resizeMode={'contain'}
        testID={'logo-img'}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#ffffff',
  },
  img: {
    width: 128,
    height: 128,
  },
});

export default Splash;
