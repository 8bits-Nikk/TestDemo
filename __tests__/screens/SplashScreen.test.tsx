import React from 'react';
import Splash from '../../src/screens/Splash/Splash';
import {renderWithNavigatorAndRedux} from '../../helpers/renderWithNavigatorAndRedux';
import {act, cleanup, screen} from '@testing-library/react-native';

const mockDispatch = jest.fn().mockImplementation(args => args);
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    dispatch: mockDispatch,
    goBack: jest.fn(),
    navigate: jest.fn(),
  }),
}));

describe('<Splash/> test', () => {
  afterEach(() => {
    cleanup();
  });

  test('Render Correctly', () => {
    const tree = renderWithNavigatorAndRedux(<Splash />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('render Image View', () => {
    renderWithNavigatorAndRedux(<Splash />);
    const imgView = screen.queryByTestId('logo-img');
    expect(imgView?.type).toBe('Image');
    expect(imgView?.props.source).not.toBe(null);
  });

  test('navigation', () => {
    jest.useFakeTimers();
    renderWithNavigatorAndRedux(<Splash />);
    act(() => {
      jest.runAllTimers();
    });
    expect(mockDispatch).toHaveBeenCalledWith({
      payload: {name: 'HomeScreen', params: undefined},
      type: 'REPLACE',
    });
  });
});
