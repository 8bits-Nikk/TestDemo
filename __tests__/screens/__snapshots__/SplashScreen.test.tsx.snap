// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash/> test Render Correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/chat.png",
      }
    }
    style={
      Object {
        "height": 128,
        "width": 128,
      }
    }
    testID="logo-img"
  />
</View>
`;
